singularity-eos/eos/eos_davis.hpp

```diff
@@ -328,7 +354,7 @@ class DavisProducts : public EosBase<DavisProducts> {
 
 PORTABLE_INLINE_FUNCTION Real DavisReactants::Ps(const Real rho) const {
   using namespace math_utils;
-  const Real y = 1.0 - _rho0 / rho;
+  const Real y = 1.0 - robust::ratio(_rho0, rho);
```

Comment:
Also see comment below.  If rho is slightly negative, y will tend to $+\infty$.  Is this a problem?  If rho is zero, then the value should be `-phat`, but it will be slightly not that (I doubt it matters unless there is some kind of numerical integration scheme being used on this routine).

Reply:
You could enforce `rho` positivity by doing `robust::ratio(_rho0, std::max(0, rho));` or something similar